polyply/src/gen_seq.py

```diff
@@ -87,21 +102,56 @@ def __init__(self, string):
         self.name = input_params[0]
         self.levels = int(input_params[1])
         self.bfact = int(input_params[2])
+        self.res_probs = input_params[3].split(',')
+
         self.residues = []
         self.weights = []
-
-        for res_prob in input_params[3].split(','):
+        for res_prob in self.res_probs:
             name, prob = res_prob.split("-")
             self.residues.append(name)
             self.weights.append(float(prob))
 
-    def gen_graph(self, seed=None):
+    def gen_graph(self, seeds=[]):
         """
-        Generate a graph from the defnitions stored in this
-        instance.
+        Generate a graph from the definitions stored in this
+        instance a list of seeds to be used in random placement
+        may be provided.
         """
         graph = _branched_graph("dum", self.bfact, self.levels)
-        graph = _random_replace_nodes(graph, self.residues, self.weights)
+        graph = _random_replace_nodes_attribute(graph, self.residues,
+                                                self.weights, "resname", seeds)
+        return graph
+
+class MacroFile():
+    """
+    Define a macro as residue graph of a
+    vermouth molecule.
+    """
+
+    def __init__(self, name, force_field):
+        """
+        Select the appropiate molecule from
+        the force_field.
+
+        Parameters:
+        -----------
+        name: str
+        force_field: `:class:vermouth.froce_field.ForceField`
+        """
+        self.molecule = force_field.blocks[name]
+
+    def gen_graph(self, seed=None):
+        """
+        Generate a graph from the definitions stored in this
+        instance by converting a molecule to a residue graph
+        and setting the resname attribute.
+        """
+        block = make_residue_graph(self.molecule, attrs=('resid', 'resname'))
+        resnames = nx.get_node_attributes(block, 'resname')
+        graph = nx.Graph()
+        graph.add_nodes_from(block.nodes)
+        graph.add_edges_from(block.edges)
+        nx.set_node_attributes(graph, resnames, "resname")
```

Comment:
So, graph = block, except that you remove all node attributes that are not 'resname'?

Reply:
pretty much also for some reason nx I/O cannot output blocks/molecules even though they are a subclass to graph

Reply:
That is so strange. I assume you tried?

Reply:
yes of course otherwise I'd just left it as it is